lib/matplotlib/widgets.py

```diff
@@ -2007,113 +2045,231 @@ def new_axes(self, ax):
+                return
+        else:
+            # Closest to an edge handle
+            self.active_handle = self._edge_order[e_idx]
 
-        self.prev = x, y
+        # Save coordinates of rectangle at the start of handle movement.
+        self._extents_on_press = self.extents
+
+    def _contains(self, event):
+        """Return True if event is within the patch."""
+        return self.rect.contains(event, radius=0)[0]
+
+    @property
+    def vmin(self):
+        """Get the start span coordinate."""
         if self.direction == 'horizontal':
-            v = x
+            vmin = self.rect.get_x()
         else:
-            v = y
+            vmin = self.rect.get_y()
+        return vmin
+
+    @property
+    def vmax(self):
+        """Get the end span coordinate."""
+        if self.direction == 'horizontal':
+            vmax = self.vmin + self.rect.get_width()
+        else:
+            vmax = self.vmin + self.rect.get_height()
+        return vmax
+
+    @property
+    def extents(self):
+        """Return (vmin, vmax)."""
+        return self.vmin, self.vmax
+
+    @extents.setter
+    def extents(self, extents):
+        # Update displayed shape
+        self._draw_shape(*extents)
+        # Update displayed handles
+        self._edge_handles.set_data(self.extents)
+        self.set_visible(self.visible)
+        self.update()
+
+
+class ToolLineHandles:
+    """
+    Control handles for canvas tools.
+
+    Parameters
+    ----------
+    ax : `matplotlib.axes.Axes`
+        Matplotlib axes where tool handles are displayed.
+    positions : 1D array
+        Positions of control handles.
+    marker : str
+        Shape of marker used to display handle. See `matplotlib.pyplot.plot`.
+    marker_props : dict
+        Additional marker properties. See `matplotlib.lines.Line2D`.
+    """
 
-        minv, maxv = v, self.pressv
-        if minv > maxv:
-            minv, maxv = maxv, minv
+    def __init__(self, ax, positions, direction, marker_props=None,
+                 useblit=True):
+        self.ax = ax
+        self._markers = []
+        self.direction = direction
+        marker_props.update({'visible': False, 'animated': useblit})
+
+        line_fun = ax.axvline if self.direction == 'horizontal' else ax.axhline
+        self._markers = [line_fun(p, **marker_props) for p in positions]
+
+        self.artists = self._markers
+
+    @property
+    def positions(self):
+        method = 'get_xdata' if self.direction == 'horizontal' else 'get_ydata'
+        return [getattr(line, method)() for line in self._markers]
+
+    def set_data(self, positions):
+        """Set x positions of handles."""
+        method = 'set_xdata' if self.direction == 'horizontal' else 'set_ydata'
+        for line, p in zip(self._markers, positions):
+            getattr(line, method)(p)
+
+    def set_visible(self, val):
+        self._markers.set_visible(val)
+
+    def set_animated(self, val):
+        self._markers.set_animated(val)
```

Comment:
```suggestion
def set_animated(self, val):
    for m in self._markers:
        m.set_animated(val)
```

otherwise you get `AttributeError: 'list' object has no attribute 'set_animated'`